### PR TITLE
Modular HTML UI + Server swapping feature (HIGH DOWNSTREAM IMPORTANCE)

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Adds the ability to add multiple servers to a server swap function on the lobby screen - See server_ips.txt to add more.

# FOR DOWNSTREAMS:

## Server swap
To make sure your server doesn't/does have this feature, see config.txt
##Enable server swapping, uncomment to enable reading of swap_ips.txt (file directory: config/skyrat/swap_ips.txt)
SERVER_SWAP_ENABLED
if you use this, make sure swap_ips.txt is present in skyrat config.

## Lobby HTML
We offer the option to customise the lobby HTML by giving you access to a file named lobby_html.txt in the config. The server will runtime if this file does not exist. Ensure it exists in directory `config/skyrat/lobby_html.txt`.

Instructions on editing the HTML:
**DO NOT UNDER ANY CIRCUMSTANCES RENAME THE ELEMENTS WITHIN THE HTML FILE, KEEP THEM AS THEY ARE.**

Elements:
.container_nav - This is the main menu container box, it defines where the menu is, and what it looks like. 
.container_terminal - This is the startup terminal html, generally, don't change this unless you want a cooler startup terminal.
.container_notice - This is the admin notice container for when an admin sets the title notice. This button is under fun in the admin tab.
.menu_a - The "buttons" for the main menu(join, observe, etc).
.menu_a:hover - The animation for hovering over buttons.
.menu_ab - This is the new polls text, so players attention is brought to it, flashes in and out with @ pollsmove animation.
.menu_b - Terminal text
.menu_c - Admin notice text.

REMEMBER, DO NOT EDIT THESE ELEMENT NAMES ELSE THE LOBBYSCREEN WILL BREAK.

Examples of what you can do: 
image
image

## Changelog
:cl:
add: You can now swap between servers! Check out the "server swap" button on the lobby screen!
add: Admins can now set a lobby notice on the main screen.
add: A cool HTML system for downstream servers to edit their lobby screens.
/:cl:

### CODE REVIEW

**Comment by ORCACommander** [comment]
is this for players to rapidly swap between live servers, or is this a way for server admins to swap instances in tgs 4/other server management software?

**Comment by Gandalf2k15** [comment]
> is this for players to rapidly swap between live servers, or is this a way for server admins to swap instances in tgs 4/other server management software?

To swap rapidly between servers

File: modular_skyrat/modules/lobbyscreen/code/load_servers.dm [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+SUBSYSTEM_DEF(serverswap)
+	name = "Server Swap"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_DEFAULT
+
+	var/file_path
+	var/icon/startup_splash
+
+/datum/controller/subsystem/title/Initialize()
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/lobbyscreen/code/load_servers.dm]:
```suggestion
/datum/controller/subsystem/serverswap/Initialize()
```

File: modular_skyrat/modules/lobbyscreen/code/load_servers.dm [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+GLOBAL_LIST_EMPTY(swappable_ips)
+
+SUBSYSTEM_DEF(serverswap)
+	name = "Server Swap"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_DEFAULT
+
+	var/file_path
+	var/icon/startup_splash
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/lobbyscreen/code/load_servers.dm]:
```suggestion
```

File: modular_skyrat/modules/lobbyscreen/code/load_servers.dm [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/datum/config_entry/flag/server_swap_enabled
+
+GLOBAL_LIST_EMPTY(swappable_ips)
+
+SUBSYSTEM_DEF(serverswap)
+	name = "Server Swap"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_DEFAULT
+
+	var/file_path
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/lobbyscreen/code/load_servers.dm]:
```suggestion
```

**Pull request merged**